pkg/gofr/datasource/dbresolver/circuit_breaker.go

```diff
@@ -0,0 +1,55 @@
+package dbresolver
+
+import (
+	"sync/atomic"
+	"time"
+)
+
+// Circuit breaker for replica health with atomic operations.
+type circuitBreaker struct {
+	failures    atomic.Int32
+	lastFailure atomic.Int64
```

Comment:
If this is a time, then use atomic.Value

Here it looks like strange.

Maybe this

https://github.com/changkun/redir/blob/3464f00e4071fa80d9eb4a56135589d92d50f708/auth.go#L47

Or using atomic.Pointer[time.Time]